internal/subscriber/core_test.go

```diff
@@ -0,0 +1,243 @@
+	cs := getMockConsumerCore(ctx, ctrl)
+	type fields struct {
+		core *Core
+	}
+	type args struct {
+		ctx                    context.Context
+		subscriberID           string
+		subscription           *subscription.Model
+		timeoutInMs            int
+		maxOutstandingMessages int64
+		maxOutstandingBytes    int64
+		requestCh              chan *PullRequest
+		ackCh                  chan *AckMessage
+		modAckCh               chan *ModAckMessage
+	}
+	tests := []struct {
+		name     string
+		fields   fields
+		args     args
+		expected *Subscriber
+		wantErr  bool
+	}{
+		{
+			name: "Create new subscriber basicImpl",
+			fields: fields{
+				core: getCore(mockBrokerStoreCore, mockSubscriptionCore, mockOffsetCore, ch),
+			},
+			args: args{
+				ctx:                    ctx,
+				subscriberID:           subID,
+				subscription:           &sub,
+				timeoutInMs:            1000,
+				maxOutstandingMessages: 1000,
+				maxOutstandingBytes:    1000,
+				requestCh:              requestCh,
+				ackCh:                  ackCh,
+				modAckCh:               modAckCh,
+			},
+			expected: &Subscriber{
+				subscription: &sub,
+				topic:        topicName,
+				subscriberID: subID,
+				requestChan:  requestCh,
+				ackChan:      ackCh,
+				modAckChan:   modAckCh,
+			},
+			wantErr: false,
+		},
+		{
+			name: "Create new subscriber basicImpl Failure",
+			fields: fields{
+				core: getCore(mockBrokerStoreCore, mockSubscriptionCore, mockOffsetCore, ch),
+			},
+			args: args{
+				ctx:                    ctx,
+				subscriberID:           subID,
+				subscription:           &sub,
+				timeoutInMs:            1000,
+				maxOutstandingMessages: 1000,
+				maxOutstandingBytes:    1000,
+				requestCh:              requestCh,
+				ackCh:                  ackCh,
+				modAckCh:               modAckCh,
+			},
+			expected: nil,
+			wantErr:  true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockBrokerStoreCore.EXPECT().GetConsumer(gomock.Any(), gomock.Any()).
+				DoAndReturn(func(arg0 context.Context, arg1 messagebroker.ConsumerClientOptions) (messagebroker.Consumer, error) {
+					if tt.wantErr {
+						return nil, errors.New("Test Error")
+					}
+					return cs, nil
+				})
+			mockMessageBroker.EXPECT().Resume(gomock.Any(), gomock.Any()).AnyTimes().Return(nil)
+			ch.EXPECT().Get(gomock.Any(), gomock.Any()).Return([]byte{'0'}, nil).AnyTimes()
+			ch.EXPECT().Set(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+			sub, err := tt.fields.core.NewSubscriber(tt.args.ctx, tt.args.subscriberID, tt.args.subscription, tt.args.timeoutInMs, tt.args.maxOutstandingMessages, tt.args.maxOutstandingBytes, tt.args.requestCh, tt.args.ackCh, tt.args.modAckCh)
+			if err == nil {
+				assert.True(t, EqualOnly(sub, tt.expected, []string{"topic", "subscriberID", "requestChan", "ackChan", "modAckChan"}))
+			}
+			if err != nil && !tt.wantErr {
+				t.Errorf("NewSubscriber Error %v", err)
+			}
```

Comment:
what if the wantErr is true, and subscriber didn't return any error..?

Reply:
added check for that as well